### ISSUE TITLE
ytt support a toml library

### ISSUE THREAD

**Issue by voor** [k14s/ytt#259]
**Describe the problem/challenge you have**
Starting to see more instances of toml showing up as kubernetes or similar snippets embedded inside yaml:

https://github.com/containerd/containerd/blob/master/docs/man/containerd-config.toml.5.md
https://docs.gitlab.com/runner/configuration/advanced-configuration.html

**Describe the solution you'd like**
Add support for:

```
load("@ytt:toml", "toml")

toml.encode({"a": [1,2,3,{"c":456}], "b": "str"})
toml.decode('a = [1, 2, 3, { c = 456 }] \nb = "str")
```

**Anything else you would like to add:**
Might be helpful?
https://github.com/sclevine/yj/blob/master/toml/encoder.go
https://github.com/sclevine/yj/blob/master/toml/decoder.go

**Comment by danielhelfand** [comment]
Thanks for opening this @voor. I think this sounds reasonable for us to further consider. Any particular immediate use case for this, or just generally you have seen toml embedded in YAML and wanted to raise this?

**Comment by voor** [comment]
We're doing advanced configuration for gitlab runners, which is managed in toml.  This is currently being done with yaml-templated-strings but it's messy and error prone.

I am aware of other users that have some echo scripts into containerd toml configuration, which is very brittle and would be better implemented with a ytt overlay.

**Comment by pivotaljohn** [comment]
Thanks to @sestegra, this is done. To be included in the next release.

**Comment by cari-lynn** [comment]
Available in [v0.38.0](https://github.com/vmware-tanzu/carvel-ytt/releases/tag/v0.38.0)
